moduleframework/mtf_scheduler.py

```diff
@@ -229,21 +248,50 @@ def __init__(self, args, unknown):
         core.print_debug("additionalAvocadoArg = {0}".format(
             self.additionalAvocadoArg))
 
+    def check_tests(self):
+        summary_line = "TEST TYPES SUMMARY"
+        prefix_line = "Type"
+        output = subprocess.check_output([self.AVOCADO, "list", "-V", "--"] + self.tests)
+        assert summary_line in output
+        badstates = ["NOT_A_TEST", "MISSING", "ACCESS_DENIED", "BROKEN_SYMLINK"]
+        badtests = []
+        # remove header line and remove last lines with is test types summary
+        testlines = []
+        for line in output.split("\n"):
+            if line.startswith(prefix_line) or not line.strip():
+                continue
+            elif line.startswith(summary_line):
+                break
+            else:
+                testlines.append(line.strip())
+        for testline in testlines:
```

Comment:
This can be done inside `for line in output.split("\n")` cycle. New cycle is not needed.

Reply:
makes sense, thanks.